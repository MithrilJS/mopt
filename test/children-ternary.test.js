"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("Ternary expressions", () => {
        it("should convert when all entries are literals (strings)", () =>
            expect(
                code(`m("0", foo ? "1" : "2")`)
            )
            .toMatchSnapshot()
        );

        it("should convert when all entries are literals (string : number)", () =>
            expect(
                code(`m("0", foo ? "1" : 2)`)
            )
            .toMatchSnapshot()
        );

        it("should convert nested ternaries when all entries are literals (strings)", () =>
            expect(
                code(`m("0", foo ? "1" : bar ? "2" : "3")`)
            )
            .toMatchSnapshot()
        );
        
        it("should not convert when entries are not literals", () => {
            // Can't convert this, dunno what `bar` is
            expect(
                code(`m("0", foo ? bar : "baz")`)
            )
            .toMatchSnapshot();
            
            // Not a text child, so can't safely optimize
            expect(
                code(`m("0", foo ? { class : options.class } : null)`)
            )
            .toMatchSnapshot();
        });
    });
});
