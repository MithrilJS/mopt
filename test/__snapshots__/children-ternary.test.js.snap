// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children Ternary expressions should convert nested ternaries when all entries are literals (strings) 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,foo?\\"1\\":bar?\\"2\\":\\"3\\",undefined);"`;

exports[`Children Ternary expressions should convert when all entries are literals (string : number) 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,foo?\\"1\\":2,undefined);"`;

exports[`Children Ternary expressions should convert when all entries are literals (strings) 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,foo?\\"1\\":\\"2\\",undefined);"`;

exports[`Children Ternary expressions should not convert when entries are not literals 1`] = `"m(\\"0\\",foo?bar:\\"baz\\");"`;

exports[`Children Ternary expressions should not convert when entries are not literals 2`] = `"m(\\"0\\",foo?{class:options.class}:null);"`;
